packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -343,12 +348,33 @@ class TextPainter {
     }
   }
 
+  bool _lastLayoutWasToMaxIntrinsicWidth = false;
+
   /// Computes the visual position of the glyphs for painting the text.
   void layout() {
     if (!_needsLayout)
       return;
     _paragraph.layout();
     _needsLayout = false;
+    _lastLayoutWasToMaxIntrinsicWidth = false;
+  }
+
+  /// Computes the visual position of the glyphs using the unconstrainted max intrinsic width.
+  void layoutToMaxIntrinsicWidth() {
```

Comment:
Doesn't work for RTL text.  It will be positioned from the right edge of the width instead of the left edge.